### PR TITLE
add auto publish workflow on new tag, release

### PR DESCRIPTION
**Pull request by loin3**
currently It is not clear which npm token to use. 
This PR should be merged after setting token name in github secrets.

### CODE REVIEW

**Review by zemyblue — COMMENTED** [review]
Please add changelog

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: "14"
+          registry-url: 'https://registry.npmjs.org'
+      - run: if [ ! -x "$(command -v yarn)" ]; then npm install -g yarn; fi
```
Comment (zemyblue) [inline .github/workflows/publish.yml:22]: How about checking yarn version after install yarn?

Comment (loin3) [reply .github/workflows/publish.yml:22]: In the setup node action, yarn is installed by default. That's why it checks if there is no yarn.

**Review by jaeseung-bae — APPROVED** [review]

**Comment by dudong2** [comment]
The changelog seems to have an 80 character auto-newline. Was there any discussion about changing the format?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,12 +7,20 @@ and this project adheres to
...
+- [\#77](https://github.com/Finschia/finschia-js/pull/77) @lbmjs/finschia:
+  Enable amino sign store msgs in `wasm` and `wasmplus`
+- [\#80](https://github.com/Finschia/finschia-js/pull/80) add
+  MsgInstantiateContract2
 - [\#81](https://github.com/Finschia/finschia-js/pull/81) add git PR template
-- [\#87](https://github.com/Finschia/finschia-js/pull/87) @lbmjs/finschia: add tests for grpc errors in collection, token
+- [\#78](https://github.com/Finschia/finschia-js/pull/78) add auth publish ci on
+  new tag, release
+- [\#87](https://github.com/Finschia/finschia-js/pull/87) @lbmjs/finschia: add
+  tests for grpc errors in collection, token
```
Comment (zemyblue) [inline CHANGELOG.md:23]: Please change to oneline.

Comment (loin3) [reply CHANGELOG.md:23]: Do we have rule that changelog has to be one line?
prettier reformat every files in finschia-proto so If we want changelog to be one line, changelog file should be included in prettier ignore and changelog will not be reformatted on every circumstances.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (zemyblue) [reply .github/workflows/publish.yml:22]: Sometimes,  yarn or npm installation is not possible even though I requested it.But it's OK.

File: CHANGELOG.md (hunk 2 above)
Comment (loin3) [reply CHANGELOG.md:23]: I got it.
Then I will add `prettier ignore` and make changelog to be one lined.
but can I do it in another PR like v0.8.0 release PR?

**Review by zemyblue — APPROVED** [review]

**Pull request merged**